src/index.js

```diff
@@ -1,7 +1,26 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import App from './App';
+import { BrowserRouter as Router } from 'react-router-dom';
+import { Provider } from 'react-redux'; // eslint-disable-line no-unused-vars
+import thunk from 'redux-thunk';
+import { createStore, compose, applyMiddleware } from 'redux';
+import reducer from './reducers/index';
+import App from './containers/App';
 import registerServiceWorker from './registerServiceWorker';
+import 'bootstrap/dist/css/bootstrap.css';
 
-ReactDOM.render(<App />, document.getElementById('root'));
-registerServiceWorker();
+const createStoreWithMiddleware = compose(applyMiddleware(thunk))(createStore);
+const store = createStoreWithMiddleware(reducer);
+
+// In order to make Router test easy,
+// Separating <Router /> and <Route />.
+// See: src/__tests__/components/App.js
+ReactDOM.render(
+  <Provider store={store}>
+    <Router>
```

Comment:
The reason why I separate `Router` and `Route` is because it's more testable.
https://github.com/duyoji/oni-on-cheese/pull/53/files#diff-c3260268c8b5ce5133999899fe1332ac

In test for src/components/App.js, We can replace `Router` with `MemoryRouter` and because of that, we can check each behavior of paths. (Mock test.)